store/snapshots/types/snapshotter.go

```diff
@@ -11,6 +11,9 @@ type Snapshotter interface {
 	// Snapshot writes snapshot items into the protobuf writer.
 	Snapshot(height uint64, protoWriter protoio.Writer) error
 
+	// AnnounceSnapshotHeight informs the underlying system of a snapshot being initiated at a given height.
+	AnnounceSnapshotHeight(height int64)
```

Comment:
How can we verify that this does not break anyone?  Are there any other implementations of store that this would not work on?

Perhaps we can keep the Store interface and add a StoreWithAnnounce interface for the best possible compatibility

Reply:
There an be custom implementations in other projects. I have introduced a new interface for the announcement.